### PR TITLE
Add v3 `Graph` nodes methods

### PR DESCRIPTION
**Pull request by decentralion**
This commit implements and tests the following methods on the `Graph`:
- `addNode`
- `removeNode`
- `hasNode`
- `nodes`

Test plan: Unit tests are included.

Paired with @wchargin

### CODE REVIEW

File: src/v3/core/graph.test.js [hunk 1]
```diff
@@ -37,5 +37,60 @@ describe("core/graph", () => {
...
+    describe("node methods", () => {
+      const p = Graph.prototype;
+      const nodeMethods = [p.addNode, p.removeNode, p.hasNode];
+
+      describe("error on", () => {
+        function rejectsEdge(f) {
+          it(`${f.name} rejects EdgeAddress`, () => {
+            const e = Address.edgeAddress(["foo"]);
+            // $ExpectFlowError
+            expect(() => f.call(new Graph(), e)).toThrow("EdgeAddress");
```
Comment (wchargin) [inline src/v3/core/graph.test.js]: Can the expected message be `expected NodeAddress` or `got EdgeAddress`?
As it stands, if we had copy-pasted the error message from a node method
to an edge method, the test wouldn’t catch this.

File: src/v3/core/graph.test.js [hunk 2]
```diff
@@ -37,5 +37,60 @@ describe("core/graph", () => {
...
+          });
+        }
+        nodeMethods.forEach(graphRejectsNulls);
+        nodeMethods.forEach(rejectsEdge);
+      });
+
+      describe("work on ", () => {
```
Comment (wchargin) [inline src/v3/core/graph.test.js]: `s/on /on/`

File: src/v3/core/graph.test.js [hunk 3]
```diff
@@ -37,5 +37,60 @@ describe("core/graph", () => {
...
+        const n1 = Address.nodeAddress(["foo"]);
+        it("a graph with no nodes", () => {
+          const graph = new Graph();
+          expect(graph.hasNode(n1)).toBe(false);
+          expect(Array.from(graph.nodes())).toHaveLength(0);
+        });
+        it("a graph with a node added", () => {
```
Comment (wchargin) [inline src/v3/core/graph.test.js]: Let us check idempotency, please.

File: src/v3/core/graph.test.js [hunk 4]
```diff
@@ -37,5 +37,60 @@ describe("core/graph", () => {
...
+          const graph = new Graph().addNode(n1).removeNode(n1);
+          expect(graph.hasNode(n1)).toBe(false);
+          expect(Array.from(graph.nodes())).toHaveLength(0);
+        });
+        it("a graph with two nodes", () => {
+          const n2 = Address.nodeAddress([""]);
+          const graph = new Graph().addNode(n1).addNode(n2);
+          expect(graph.hasNode(n1)).toBe(true);
+          expect(graph.hasNode(n2)).toBe(true);
+          expect(Array.from(graph.nodes()).sort()).toEqual([n2, n1]);
```
Comment (wchargin) [inline src/v3/core/graph.test.js]: Rest in pieces, `sortedByAddress`. :100: :taco:

File: src/v3/core/graph.test.js (hunk 3 above)
Comment (decentralion) [reply src/v3/core/graph.test.js]: Done.

**Pull request merged**